subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/release/tasks/ReleaseNeeded.java

```diff
@@ -15,43 +16,57 @@
     //For reference, see the ".travis.yml" of Mockito project
     private final static String SKIP_RELEASE_ENV = "SKIP_RELEASE";
     private final static String SKIP_RELEASE_KEYWORD = "[ci skip-release]";
+    private static final String FORCE_RELEASE_KEYWORD = "[ci force-release]";
 
     public boolean releaseNeeded(ReleaseNeededTask task) {
-        return releaseNeeded(task, new EnvVariables());
+        Pair<Boolean, String> pair = releaseNeeded(task, new EnvVariables());
+
+        boolean releaseNeeded = pair.getLeft();
+        String message = pair.getRight();
+
+        if (!releaseNeeded && task.isExplosive()) {
+            throw new GradleException(message);
+        } else {
+            LOG.lifecycle(message);
+        }
+        return releaseNeeded;
     }
 
-    public boolean releaseNeeded(ReleaseNeededTask task, EnvVariables envVariables) {
+    Pair<Boolean, String> releaseNeeded(ReleaseNeededTask task, EnvVariables envVariables) {
         boolean skipEnvVariable = envVariables.getNonEmptyEnv(SKIP_RELEASE_ENV) != null;
         LOG.lifecycle("  Environment variable {} present: {}", SKIP_RELEASE_ENV, skipEnvVariable);
 
         boolean commitMessageEmpty = task.getCommitMessage() == null || task.getCommitMessage().trim().isEmpty();
         boolean skippedByCommitMessage = !commitMessageEmpty && task.getCommitMessage().contains(SKIP_RELEASE_KEYWORD);
-        LOG.lifecycle("  Commit message to inspect for keyword '{}': {}",
-                SKIP_RELEASE_KEYWORD,
-                commitMessageEmpty ? "<unknown commit message>" : "\n" + task.getCommitMessage());
+        boolean forcedByCommitMessage = !commitMessageEmpty && task.getCommitMessage().contains(FORCE_RELEASE_KEYWORD);
+        LOG.lifecycle("  Commit message to inspect for keywords '{}' and '{}': {}",
+            SKIP_RELEASE_KEYWORD, FORCE_RELEASE_KEYWORD,
+            commitMessageEmpty ? "<unknown commit message>" : "\n" + task.getCommitMessage());
 
         boolean releasableBranch = task.getBranch() != null && task.getBranch().matches(task.getReleasableBranchRegex());
         LOG.lifecycle("  Current branch '{}' matches '{}': {}", task.getBranch(), task.getReleasableBranchRegex(), releasableBranch);
 
-        ComparisonResults results = new ComparisonResults(task.getComparisonResults());
-        boolean publicationsIdentical = results.areResultsIdentical();
-        LOG.lifecycle(results.getDescription());
+        if (releasableBranch) {
+            if (skippedByCommitMessage) {
+                return Pair.of(false, " Skipping release due to skip release keyword in commit message.");
+            } else if (skipEnvVariable) {
+                return Pair.of(false, " Skipping release due to skip release env variable.");
+            } else if (task.isPullRequest()) {
+                return Pair.of(false, " Skipping release due to is PR.");
+            } else if (forcedByCommitMessage) {
```

Comment:
There are 2 problems with this approach:

a) It seems that "ci-force-release" does not quite force the release at all times. Effectively it bypasses comparison of the binaries. Hence the naming of this commit message keyword is confusing.
b) It is possible to include both "ci skip-release" and "ci force-release" in the commit message leading to somewhat surprising behavior.

Suggestions:

a) make forcing really forcing, regardless of other conditions
b) rename keyword to "[ci skip-compare-publications]" (we already have "compare-publications" and "comparePublications" plugin and tasks).

I really like b) - it is effectively your original idea :) It gives finer control. It is exactly the use case Mockito needs. It is also simpler, we don't have to worry about conflicting keywords.

Bonus - it would be cool to avoid executing the comparePublications task if the keyword is used. Not sure how easy it is to implement. I suggest to consider it in a separate PR.